.github/workflows/build_library.yml

```diff
@@ -0,0 +1,27 @@
+name: Binary Release Build
+
+on:
+  push:
+    branches:
+      - release/**
+    
+
+jobs:
+  linux:
+    name: Linux
+    runs-on: ubuntu-latest
+  
+    steps:
+      - uses: actions/checkout@v2
+    
+      - name: Build in Docker (x86-64)
+        # Add more of these steps to support other architectures
+        run: |
+          docker build --build-arg DOCKER_ARCH=amd64 -t getsentry/postgres_wal2json:build --target build .
+          docker create --name=postgres_wal2json getsentry/postgres_wal2json:build
+          docker cp postgres_wal2json:/usr/local/lib/postgresql/wal2json.so ./wal2json-Linux-x64.so
```

Comment:
This is an anti-pattern. Let's revisit this to make it more canonical and efficient. See what we do over at getsentry/sentry as an example: https://github.com/getsentry/sentry/blob/master/docker/cloudbuild.yaml

We use a `builder` image that rarely changes and then _run_ that builder image, mounting the current working directory into it, which avoids the `docker create && docker cp` dance.

Reply:
I will try to mount the working directory, but I am not sure it will work. `make install` saves the library file in a postgres directory that already contains other files.

Specifically which part is the anti pattern? Is it about using the copy? If yes what is the problem with that ?

Reply:
I added a separate Dockerfile as a builder, this should never change, then I start that container and run the build process on that. Is that what you meant ?

Reply:
>Specifically which part is the anti pattern? Is it about using the copy? If yes what is the problem with that ?

Anti-pattern is using `docker build` to generate the artifact (and of course, using the create/cp pattern afterwards to extract the artifact).

>make install saves the library file in a postgres directory that already contains other files.

We can use the `cp` inside the container as you did in your updated version.

Check the pattern over at Sentry if you wanna see an example: 

- https://github.com/getsentry/sentry/blob/a7124970a225eee4a91092b4e6620d809418b4a2/docker/cloudbuild.yaml#L2-L19
- https://github.com/getsentry/sentry/blob/master/docker/builder.dockerfile
- https://github.com/getsentry/sentry/blob/master/docker/builder.sh

Reply:
Thanks, that makes sense.
Updated according to your last comments.